raiden/network/proxies/user_deposit.py

```diff
@@ -577,3 +658,161 @@ def _deposit_check_result(
                     raise RaidenRecoverableError(msg)
 
                 raise RaidenRecoverableError("Deposit failed of unknown reason")
+
+    def _plan_withdraw_preconditions(
+        self, amount_to_plan_withdraw: TokenAmount, given_block_identifier: BlockIdentifier
+    ) -> None:
+        """ Check if a WithdrawPlan for the given amount can be created. """
+        if amount_to_plan_withdraw <= 0:
+            raise BrokenPreconditionError("Planned withdraw amount must be greater than zero.")
+
+        try:
+            current_balance = self.get_total_deposit(
+                address=self.node_address, block_identifier=given_block_identifier
+            )
+        except ValueError:
+            # If `given_block_identifier` has been pruned we can't perform the check
+            return
+        except BadFunctionCallOutput:
+            raise_on_call_returned_empty(given_block_identifier)
+
+        if current_balance < amount_to_plan_withdraw:
+            raise BrokenPreconditionError(
+                f"Can't create WithdrawPlan for amount {amount_to_plan_withdraw}, "
+                f"it exceeds the current balance of {current_balance}."
+            )
+
+    def _withdraw_preconditions(
+        self, amount_to_withdraw: TokenAmount, given_block_identifier: BlockIdentifier
+    ) -> None:
+        try:
+            withdraw_plan = self.get_withdraw_plan(
+                withdrawer_address=self.node_address, block_identifier=given_block_identifier
+            )
+            whole_balance = self.whole_balance(block_identifier=given_block_identifier)
+        except ValueError:
+            # If 'given_block_identifier' has been pruned, we can't perform the check
+            return
+        except BadFunctionCallOutput:
+            raise_on_call_returned_empty(given_block_identifier)
+
+        if amount_to_withdraw > withdraw_plan.withdraw_amount:
+            raise BrokenPreconditionError(
+                f"Can't withdraw {amount_to_withdraw}, "
+                f"current withdraw plan only allows for {withdraw_plan.withdraw_amount}."
+            )
+
+        given_block_number = self.client.get_block(given_block_identifier)["number"]
```

Comment:
I know this does not raise for a pruned block: https://github.com/raiden-network/raiden/blob/007f5453d05efe3b47b090bbe8f6f4a21b83bd2b/raiden/tests/integration/rpc/assumptions/test_parity_rpc_assumptions.py#L87

but I don't know what it does if the `given_block_identifier` was a pruned uncle. We should check if it returns `None` or if it raises.